### PR TITLE
Refactor planting-sites fetch to use redux + initial map/list view in planting sites details page

### PR DESCRIPTION
**Pull request by karthikbtf**
- was working on the tv2 updates for planting sites and had a need to fetch planting sites list
- realized we are already overfetching (in the App, in the Observations page, etc.)
- update the App to dispatch a redux request
- all components now rely on the dispatched request in the App to select planting sites
- also, when we update (edit/create) planting sites, the app reloads planting sites
- simplifies code
- added initial map/list view for planting site details page

Terraware App 🔊 2023-07-10 13-27-02

Terraware App 🔊 2023-07-10 13-26-44

### CODE REVIEW

File: src/components/PlantingSites/index.tsx [hunk 1]
```diff
@@ -16,27 +19,34 @@ export type PlantingSitesProps = {
...
-
-  // show spinner while initializing data
-  if (initializingData) {
-    return <CircularProgress sx={{ margin: 'auto' }} />;
-  }
-
-  return <PlantingSitesWrapper {...props} />;
+export default function PlantingSites({ reloadTracking }: PlantingSitesProps): JSX.Element {
+  return (
+    <Switch>
```
Comment (charlie-tf) [inline src/components/PlantingSites/index.tsx:24]: Child Switches appear to route on the full path? That is, does the app ultimately see a flattened list of Routes, or does the nesting affect how things are routed?

Comment (charlie-tf) [reply src/components/PlantingSites/index.tsx:24]: If the `PlantingSitesWrapper`, for instance, is not rendered, can the app see its routes?

**Review by charlie-tf — COMMENTED** [review]
Looks good! Thank you for this refactor. Have a couple questions about routing, as the nesting of `Switch`s confuses me a little bit.

File: src/components/PlantingSites/index.tsx (hunk 1 above)
Comment (karthikbtf) [reply src/components/PlantingSites/index.tsx:24]: That is correct, the routing is on the full path.
The root App.tsx has a route `/planting-sites` (non-exact match, just needs to start with that) - which renders the PlantingSitesWrapper. Once that is rendered, the finer route handling happens.
Not sure if I answered the question?

Comment (charlie-tf) [reply src/components/PlantingSites/index.tsx:24]: I think I understand. So a Switch isn't like a conditional renderer. It renders down its component tree to make sure all the routes are rendered?

Comment (karthikbtf) [reply src/components/PlantingSites/index.tsx:24]: The routing happens at run-time, there is no dictionary of routes to components.

Comment (karthikbtf) [reply src/components/PlantingSites/index.tsx:24]: Switch happens at run-time

**Review by charlie-tf — APPROVED** [review]

File: src/components/PlantingSites/index.tsx (hunk 1 above)
Comment (charlie-tf) [reply src/components/PlantingSites/index.tsx:24]: Thanks Karthik

Comment (karthikbtf) [reply src/components/PlantingSites/index.tsx:24]: I think I may have misunderstood the question completely - let me know if that's the case.

Comment (charlie-tf) [reply src/components/PlantingSites/index.tsx:24]: Oh it's okay. I've worked out what was confusing me.

Comment (karthikbtf) [reply src/components/PlantingSites/index.tsx:24]: No worries. I just matched a lot of your questions to the [docs](https://reactrouter.com/en/main/start/concepts). Those were good questions.
Re: routing with full paths versus relative-links, react has support for both -  we need to look for paths that don't start with `/` for relative links.. we don't do that here.. we could certainly do that though. Our route configs match the entire path starting with leading `/`.
Re: rendering routes (creating a tree, etc.) to match the paths, that happens during the render of the component. When App.tsx is rendered, the route config for App.tsx alone is processed, converted to a tree and matched.
When PlantingSitesWrapper is rendered, the same thing happens for its specific route config.
That is my understanding at least :)

**Pull request merged**